LayoutTests/fast/dom/gc-abort-controller.html

```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Test AbortController memory leak with event listeners</title>
+    <script src="../../resources/js-test.js"></script>
+</head>
+<body>
+<script>
+jsTestIsAsync = true;
+
+if (!window.testRunner || !window.internals) {
+    testFailed("Requires testRunner and internals.");
+    finishJSTest();
+}
+
+function runTest() {
+    let weakRefs = [];
+    let controllers = [];
+    
+    // Create 500 AbortControllers with event listeners
+    for (let i = 0; i < 500; i++) {
+        let controller = new AbortController();
+        controller.signal.addEventListener('abort', () => {
+            console.log('Event listener for controller ' + i);
+        });
+        controllers.push(controller);
+        weakRefs.push(new WeakRef(controller.signal));
+    }
+    
+    controllers = null;
+    
+    var cnt = 0;
+    let handle = setInterval(() => {
+        gc();
+        cnt++;
+        
+        // Count how many signals are still alive
+        let aliveCount = 0;
+        for (let weakRef of weakRefs) {
+            if (weakRef.deref()) {
+                aliveCount++;
+            }
+        }
+        
+        if (aliveCount === 0) {
```

Comment:
Our GC is conservative. As a result, we never write GC tests that assume all objects get collected. We write tests that *some* objects get collected.